### DISCUSSION TITLE
Is there a setting to enable hot reloading when the json files change?

### DISCUSSION THREAD

**Discussion by looeee** [amannn/next-intl#1200] [General]
Similar to [reloadOnPrerender ](https://github.com/i18next/next-i18next?tab=readme-ov-file#reloading-resources-in-development) in next-i18next.

**Comment by looeee** [comment]
I was just about to give up and move on to doing some actual translation work when I saw the comment in the docs version of i18n.ts: *When using Turbopack, this will enable HMR for `en`*

``` ts
export default getRequestConfig(async ({ locale }) => {
  if (!locales.includes(locale as any)) notFound();

  return {
    messages: (
      await (locale === "en"
        ? // When using Turbopack, this will enable HMR for `en`
          import("./messages/en.json")
        : import(`./messages/${locale}.json`))
    ).default,
  };
});
```

Sure enough, `npm run dev --turbo` makes the English translation hot reload 🥳 
I don't have enough experience to know whether `--turbo` this will cause other problems but hopefully it's a solution.

**Comment by parsa-vxt** [comment]
Had the same issue. Moving my message files to the `src` folder fixed the hot reload.
